arbor/include/arbor/mechanism.hpp

```diff
@@ -68,6 +70,21 @@ class mechanism {
     arb_mechanism_interface iface_;
     arb_mechanism_ppack ppack_;
     arb_value_type** time_ptr_ptr = nullptr;
+
+private:
+#ifdef ARB_PROFILE_ENABLED
```

Comment:
This seems like these two snippets would be useful elsewhere. I would therefore suggest moving them to `arb::profile`.

However, I am a wary of the `static`, it means that `id` is initialised once and only the first region is tagged. See here for a minimal example
https://gcc.godbolt.org/z/f3GPeWaW6

Therefore my final suggestion would be this
```c++
#ifdef ARB_PROFILE_ENABLED
#define PROFILE_ENTER(x) do { static auto id = arb::profile::profiler_region_id(x); arb::profile::profiler_enter(id); } while(0)
#else
#define PROFILE_ENTER(x) do {} while(0)
#endif
```

Reply:
You're right about static, all the instances of the `mechanism` class end up having the same name, and the profiler doesn't work. Does your suggestion actually fix this? I went a different route.

Reply:
Since it basically pastes in the `static id =` once per invocation, it must fix it, yes.